pkg/kubelet/cm/devicemanager/topology_hints.go

```diff
@@ -41,6 +44,18 @@ func (m *ManagerImpl) GetTopologyHints(pod v1.Pod, container v1.Container) map[s
 				continue
 			}
 
+			allocated := m.getAllocatedDevices(string(pod.UID), container.Name, resource)
+			if allocated.Len() > 0 {
```

Comment:
Why in this case do we send back nil hints? Or in what case can this occur?

Reply:
It's not exactly `nil` hints, its an empty list of hints, which the `TopologyManager` interprets differently than `nil`. It's the way of encoding an "error" in the sense that no valid hints could be generated by this hint provider.

`nil` == I don't care about affinity, do whatever you want with this information
`[]topologymanager.TopologyHint{}` == I care about affinity, and I have no hints to give

That said, yeah, this case should ever actually come up in practice. Could even be replaced by a panic if we wanted.

Reply:
Oh okay if its interpreted as an error then that makes sense to me. Thanks for the clarification!